app/forms.py

```diff
@@ -27,6 +28,11 @@ def validate_email(email):
         if user:
             raise ValidationError('This email is already taken')
 
+    def validate_army_name(self, army_name):
+        for letter in army_name.data:
+            if not (letter >= 'a' and letter <= 'z') or (letter >= 'A' and letter <= 'Z'):

```

Comment:
There's a function called `isalphanumeric` that you can do on strings.
This will allow the user to use letters and numbers